src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2252,43 +2283,51 @@ private bool UseUpdatedEscapeRulesForInvocation(Symbol symbol)
             {
                 foreach (var (_, argument, refKind) in escapeArguments)
                 {
+                    if (ShouldInferDeclarationExpressionValEscape(argument, out _))
```

Comment:
I think we also wanted to change this to use 'isMixableArgument'.

I think the behavior/factoring for the current release is going to be fine, we just want to make sure to iron these kinds of things out when we refactor in 17.5.